SwiftMessages/Animator.swift

```diff
@@ -65,4 +65,14 @@ public protocol Animator: class {
     func show(context: AnimationContext, completion: @escaping AnimationCompletion)
 
     func hide(context: AnimationContext, completion: @escaping AnimationCompletion)
+
+    var showDuration: TimeInterval? { get }
+
+    var hideDuration: TimeInterval? { get }
```

Comment:
Why Optional? Why not a default value of 0? What does `nil` mean in this case?

Reply:
`nil` would mean that the duration is unknown, such as if the animation was being provided by `UIDynamicAnimator`.

Reply:
I think it could be helpful to explain that in a comment, since it's a public API and not immediately obvious (at least to me).